traceapp/app.go

```diff
@@ -45,17 +46,35 @@ type App struct {
 	tmplLock sync.Mutex
 	tmpls    map[string]*htmpl.Template
 
-	Log *log.Logger
+	Log     *log.Logger
+	baseURL *url.URL
 }
 
 // New creates a new application handler. If r is nil, a new router is
 // created.
-func New(r *Router) *App {
+//
+// The given base URL is the absolute base URL under which traceapp is being
+// served, e.g., "https://appdash.mysite.com" or "https://mysite.com/appdash".
+// The base URL must contain a scheme and host, or else an error will be
+// returned.
+func New(r *Router, base *url.URL) (*App, error) {
 	if r == nil {
 		r = NewRouter(nil)
 	}
 
-	app := &App{Router: r, Log: log.New(os.Stderr, "appdash: ", log.LstdFlags)}
+	// Validate the base URL and use the root path if none was specified.
+	if base.Scheme == "" || base.Host == "" {
+		return nil, fmt.Errorf("appdash: base URL must contain both scheme and port, found %q", base.String())
```

Comment:
you are checking host is non-nil, but then complain about the port missing.

Also, can we not just trust that the user passed in a reasonable URL? We don't actually use the scheme or host specifically right?